modules/objectdb_profiler/editor/objectdb_profiler_panel.h

```diff
@@ -0,0 +1,100 @@
+/**************************************************************************/
+/*  objectdb_profiler_panel.h                                             */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef OBJECTDB_PROFILER_PANEL_H
+#define OBJECTDB_PROFILER_PANEL_H
+
+#include "core/io/dir_access.h"
+#include "core/templates/lru.h"
+#include "data_viewers/snapshot_view.h"
+#include "scene/gui/tab_container.h"
+#include "scene/gui/tree.h"
+#include "snapshot_data.h"
+
+const int SNAPSHOT_CACHE_MAX_SIZE = 10;
+
+enum RC_MENU_OPERATIONS {
+	RENAME,
+	SHOW_IN_FOLDER,
+	DELETE,
```

Comment:
I'm not sure why this isn't being picked up by the CI runs, nor why this seems to compile just fine with MSVC, but `DELETE` here seems to create problems when compiling for Windows with latest mingw-w64 (GCC) from macOS Homebrew, which is version 12.0.0 as of writing this. The reason for this seems to be that `winnt.h` defines this `DELETE` token already, as seen [here](https://github.com/mingw-w64/mingw-w64/blob/99684a93993bd8288993c0f47014624523bca0cf/mingw-w64-headers/include/winnt.h#L3040).

Just giving these a more specific name would presumably resolve the problem. You'll likely also want to rename `RC_MENU_OPERATIONS`, since it doesn't confirm to the convention of PascalCase for enums in Godot. I would probably also suggest moving this (and the `PartialSnapshot` struct) into the `ObjectDBProfilerPanel` class so as to not pollute the global namespace, but that's a bit of a nitpick I suppose.